openmaptiles/styleutils.py

```diff
@@ -0,0 +1,87 @@
+import yaml
+import json
+from pathlib import Path
+from openmaptiles.tileset import Tileset
+
+
+def fp_to_dict(fp: str) -> dict:
+    with open(fp, 'r') as f:
+        d = json.load(f)
+    return d
+
+
+def get_ts_lyr_style_json_fp(yaml_fp: str) -> Path:
+    yaml_path = yaml_fp.filename
+    json_name = Path(yaml_path.name).with_suffix('.json')
+    ts_lyr_dir = yaml_path.parent
+
+    return ts_lyr_dir.joinpath(json_name)
+
+
+def add_order(lyrs: list) -> list:
+    for i in range(0, len(lyrs)):
+        lyr = lyrs[i]
+        lyr['order'] = i
+    return lyrs
+
+
+def get_order(l: dict) -> int:
+    return int(l.get('order'))
+
+
+def split(tileset_fp: Path, style_fp: Path):
+    tileset = Tileset.parse(tileset_fp)
+    tileset_lyrs = tileset.layers
+
+    # Add order to each style layer (e.g. backgroud - 0, labels - 196)
+    style = fp_to_dict(style_fp)
+    style_lyrs_w_order = add_order(style.get('layers'))
+
+    # ts_lyr is a layer in tileset.yaml (e.g. landuse, landcover, water, waterway...)
+    for ts_lyr in tileset_lyrs:
+        ts_lyr_style_json_fp = get_ts_lyr_style_json_fp(ts_lyr)
+        ts_lyr_style_lyrs = list()
+        for style_lyr in style_lyrs_w_order:
+            source_layer = style_lyr.get('source-layer')
+            if source_layer == ts_lyr.filename.stem:
+                ts_lyr_style_lyrs.append(style_lyr)
+        out_dict = {'layers': ts_lyr_style_lyrs}
+
+        # write style layers with order to json for specific tileset layer
+        with open(ts_lyr_style_json_fp, 'w') as fout:
```

Comment:
can also use `ts_lyr_style_json_fp.write_text(json.dumps(...))`
(this var name is ... complicated):

Reply:
Renamed to `snippet_fp` and `snippet_lyrs`, I hope that's easier to understand.